File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Directive.kt
package com.nouprax.markdown.core

public class Directive internal constructor(
    public val name: String,
    /** Markup owned by the label field, not a generic child/content element. */
    public val label: DirectiveLabel?,
    override val scope: Scope,
    override val anchor: String?,
    override val attributes: Attributes,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
}

File: packages/es-markdown-core/src/markup/directive.ts
import type { MarkupBase } from "./base.js";
import type { DirectiveLabel } from "./directive-label.js";

export interface Directive extends MarkupBase<"directive"> {
    readonly name: string;
    /** Markup owned by the label field, not a generic child/content element. */
    readonly label: DirectiveLabel | null;
}
